### PR TITLE
Upgrade/go 1.15

### LINKED ISSUE #1899: Migrate to using SANs for webhook certificates for Go 1.15
Context: https://github.com/googleforgames/agones/pull/1897

Go 1.15 no longer supports Common Name based certs for HTTPS without a specific flag. Common Name is a 20 year old deprecation, and we probably shouldn't be promoting.

There is a backward compatible environment flag, but it will be going away eventually. This flag, for the life of me I could not make it work -- which blocked a migration to latest 1.15.

I tried to write steps to create certs with SANs, but my OpenSSL knowledge is not good enough.

So, as a plan to migrate to Go 1.15:

1. [x] Update [webhook with cert creation](https://agones.dev/site/docs/getting-started/create-webhook-fleetautoscaler/#2-create-x509-root-and-webhook-certificates) instructions to use SANs over Common Name, and ensure it works with current Agones version.
2. [x] Provide release note announcement on deprecation of Common Name certificates for webhooks.
3. [ ] Migrate build system to Go 1.15 and no longer support SAN based certificates.

### LINKED ISSUE #2024: Can't run Allocation example with Go 1.16
**What happened**:

Attempting to run the allocation service sample with Go 1.16 results in the following error:

```
$ go run main.go --ip 35.232.217.29 --namespace default --key /home/markmandel/cloud-game-servers-examples/terraform-4-cluster/allocation/certs/gke_gcgs-beta-demo_us-central1-b_game-cluster-us-2/client.key --cert /home/markmandel/cloud-game-servers-examples/terraform-4-cluster/allocation/certs/gke_gcgs-beta-demo_us-central1-b_game-cluster-us-2/client.crt --cacert /home/markmandel/cloud-game-servers-examples/terraform-4-cluster/allocation/certs/gke_gcgs-beta-demo_us-central1-b_game-cluster-us-2/ca.crt --multicluster truepanic: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: authentication handshake failed: x509: cannot validate certificate for 35.232.217.29 because it doesn't contain any IP SANs"
goroutine 1 [running]:main.main()        /home/markmandel/agones/examples/allocator-client/main.go:76 +0x81cexit status 2
$ go version
go version go1.16 linux/amd64
```

**What you expected to happen**:

It to successfully allocate.

**How to reproduce it (as minimally and precisely as possible)**:

Follow https://agones.dev/site/docs/advanced/allocator-service/ with openssl certificates, and then attempt to allocate.

**Anything else we need to know?**:

Not sure of workaround at this stage.

**Environment**:
- Agones version: 1.12.0
- Kubernetes version (use `kubectl version`): 1.17.0
- Cloud provider or hardware configuration: GKE
- Install method (yaml/helm): Helm
- Troubleshooting guide log(s): Above
- Others: N/A

### PR DESCRIPTION
**Pull request by cindy52**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

 /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:
Upgrade go version to 1.15.13 Go 1.15 no longer supports Common Name based certs for HTTPS without a specific flag. This PR includes adding SAN for all related certs and updating the doc to create a cert with SAN for allocation service.

**Which issue(s) this PR fixes**:

Closes #1899
Closes #2024

### CODE REVIEW

**Comment by cindy52** [comment]
@markmandel , I upgraded to go 1.15 instead and create another issue #2168  to address `gen-api-docs` failing. Since go 1.15 will require SAN for certs as well, thinking it might be good to check in the adding SAN change and later fix the `gen-api-docs` issue.

**Comment by markmandel** [comment]
> @markmandel , I upgraded to go 1.15 instead and create another issue #2168 to address `gen-api-docs` failing. Since go 1.15 will require SAN for certs as well, thinking it might be good to check in the adding SAN change and later fix the `gen-api-docs` issue.

Makes sense to me 👍🏻

File: build/build-image/Dockerfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ RUN apt-get update && \
 
 # install go
 WORKDIR /usr/local
-ENV GO_VERSION=1.14.12
+ENV GO_VERSION=1.15.13
 ENV GOPATH /go
 ENV GO111MODULE=on
```
Comment (roberthbailey) [inline build/build-image/Dockerfile:28]: It looks like we will be able to get rid of this env var with go 1.16 (where on becomes the default).

File: site/content/en/docs/Advanced/allocator-service.md [hunk 2]
```diff
@@ -96,6 +96,9 @@ EOF
 # Add ca.crt to the allocator-tls-ca Secret
 TLS_CA_VALUE=$(kubectl get secret allocator-tls -n agones-system -ojsonpath='{.data.ca\.crt}')
 kubectl get secret allocator-tls-ca -o json -n agones-system | jq '.data["tls-ca.crt"]="'${TLS_CA_VALUE}'"' | kubectl apply -f -
+echo $TLS_CA_VALUE | base64 -d > ca.crt
```
Comment (roberthbailey) [inline site/content/en/docs/Advanced/allocator-service.md:99]: Are these changes only necessary once we make the next release (and are using go 1.15)? If so, they should be behind a feature gate: https://agones.dev/site/docs/contribute/#within-a-page

Comment (markmandel) [reply site/content/en/docs/Advanced/allocator-service.md:99]: I don't think they need to go behind a feature gate, as they will still work with the current release - and if people do it now, they don't have to do it later. So no issue as it is on my end.

Comment (roberthbailey) [reply site/content/en/docs/Advanced/allocator-service.md:99]: as long as they work with both old and new versions, sgtm.

Comment (cindy52) [reply site/content/en/docs/Advanced/allocator-service.md:99]: Go 1.15 no longer supports Common Name based certs for HTTPS without a specific flag.  Adding SAN to the cert is the solution and it's also backward compatible.

**Review by roberthbailey — APPROVED** [review]

**Pull request merged**